scripts/ci/gitlab/pipeline/build.yml

```diff
@@ -79,10 +79,12 @@ build-malus:
     - .compiler-info
     - .collect-artifacts
   script:
-    - time cargo build --locked --profile testnet --verbose -p polkadot-test-malus
+    - time cargo build --locked --profile testnet --verbose -p polkadot-test-malus -p polkadot-node-core-pvf-prepare-worker -p polkadot-node-core-pvf-execute-worker
     # pack artifacts
     - mkdir -p ./artifacts
     - mv ./target/testnet/malus ./artifacts/.
+    - mv ./target/testnet/polkadot-node-core-pvf-prepare-worker ./artifacts/polkadot-prepare-worker
+    - mv ./target/testnet/polkadot-node-core-pvf-execute-worker ./artifacts/polkadot-execute-worker
```

Comment:
can't we do a mv testnet/*? The name change seems non-essential.

Reply:
`malus`, as a normal node, expects auxiliary binaries to be named accordingly (their names [are set as constants](https://github.com/paritytech/polkadot/pull/7337/files#diff-a876747ba406343553d4ee5b31824f19837b394b22daf9624f1f7bc09a2077e6R55-R59)). I tried to introduce them as package binaries of `malus` package (the same way they are built with the normal node) but that results in `rust-doc` conflicts as it tries to generate documentation for effectively the same package twice. I didn't find an easy way to avoid that and decided to leave it like this as `malus` is not something supposed to be run by an end user anyway, and in our case, it's deployed to the testnet from the Docker image so it's enough to build that image properly.

Reply:
> without specifying `-p prepare-worker etc...` (after adding the bins and some dependencies to `malus/Cargo.toml`). I couldn't reproduce any issues with `cargo doc`.

Try it in CI, not locally. I wasn't able to make the `rust-doc` job pass with binaries in malus' `Cargo.toml`. Maybe the CI may be patched to make it pass, I'm not sure. Have just chosen the most straightforward way :slightly_smiling_face:

Reply:
Yep, was able to reproduce with `cargo doc --workspace --no-deps`. But I fixed it by just setting `doc = false` in malus's `Cargo.toml`.